### DISCUSSION TITLE
关于自制数据集标注文件的格式

### DISCUSSION THREAD

**Discussion by huadouxiansheng** [open-mmlab/mmselfsup#712] [Q&A]
你好，已知CustomDataset 的数据格式：
train/
├── folder_1
│   ├── xxx.png
│   ├── xxy.png
│   └── ...
├── 123.png
├── nsdf3.png
└── ...
标注文件为
folder_1/xxx.png 0
folder_1/xxy.png 1
123.png 1
nsdf3.png 2
...
请问要是自己的数据集是voc格式，所有图片在一个文件夹里（没有分类在各自类别的文件夹），标注文件怎么改成上面的样子呢。还有就是想问一下用mmselfsup训练数据集中图片上有多个目标会对训练效果产生影响吗，是不是尽量一个图片对于一个目标训练效果比较好

**Comment by fangyixiao18** [comment]
You could use ImageListDataset, you need to generate your txt file recursively

**Comment by fangyixiao18** [comment]
For your second question, the self-supervised learning algorithm which you applied also influenced the performance. 
did you appliy contrastive learning or masked image modeling?

**Reply by huadouxiansheng** [reply]
是的,使用了mocov2的方法

**Reply by fangyixiao18** [reply]
> 是的,使用了mocov2的方法

With mocov2, one object in one image is better
